api/server/server.go

```diff
@@ -20,12 +20,20 @@ import (
 
 type ToDDApi struct {
 	cfg config.Config
+	tdb db.DatabasePackage
 }
 
-func (tapi ToDDApi) Start(cfg config.Config) {
+func (tapi ToDDApi) Start(cfg config.Config) error {
 
 	tapi.cfg = cfg
 
+	tdb, err := db.NewToddDB(tapi.cfg)
+	if err != nil {
+		return err
```

Comment:
I think Dave Cheney's point was that the example could be made simpler.

The example:
``` go
err := fn()
if err != nil {
       return err
}
return nil
```

Could be written as below and achieve the exact same result:
``` go
return fn()
```

In this case we're trying to get a new instance of a DatabasePackage interface. If successful, continue, if not return the error up.